### PR TITLE
Exec related fixes

### PR DESCRIPTION
**Pull request by dimkouv**
1) fix contract reader query limit
2) use msgVisibilityInterval and stop tracking last report timestamp

### CODE REVIEW

File: execute/plugin.go [hunk 1]
```diff
@@ -172,6 +166,9 @@ func (p *Plugin) Observation(
 		}
 	}
 
+	msgVisibilityInterval := p.cfg.MessageVisibilityInterval
+	fetchFrom := time.Now().Add(-msgVisibilityInterval)
```
Comment (makramkd) [inline execute/plugin.go:170]: I'm not 100% sure about this but does this time need to be in UTC? We're comparing this to block timestamps eventually right?

File: execute/plugin_functions.go [hunk 2]
```diff
@@ -154,6 +154,31 @@ func filterOutExecutedMessages(
...
+				executedSeqNumsForReport.Add(seqNum)
+			}
+		}
+		report.ExecutedMessages = executedSeqNumsForReport.ToSlice()
+		sort.Slice(report.ExecutedMessages, func(i, j int) bool {
+			return report.ExecutedMessages[i] < report.ExecutedMessages[j]
+		})
+	}
+	// ----------------------------------------
+
```
Comment (winder) [inline execute/plugin_functions.go:181]: We need a test case for this. I'm not sure what this does differently than the original logic

File: execute/plugin.go (hunk 1 above)
Comment (makramkd) [inline execute/plugin.go:170]: Can probably inline this into the call site, unless we're using for logging somewhere

File: internal/reader/ccip.go [hunk 3]
```diff
@@ -161,7 +161,6 @@ func (r *CCIPChainReader) CommitReportsGTETimestamp(
 		},
 		query.LimitAndSort{
 			SortBy: []query.SortBy{query.NewSortByTimestamp(query.Asc)},
-			Limit:  query.Limit{Count: uint64(limit)},
```
Comment (makramkd) [inline internal/reader/ccip.go:164]: Is this limit applied somewhere in the filtering code below?

File: execute/plugin_functions.go (hunk 2 above)
Comment (dimkouv) [reply execute/plugin_functions.go:181]: not needed eventually, removing it

Comment (dimkouv) [reply execute/plugin_functions.go:181]: was fine before

**Review by winder — APPROVED** [review]
lgtm

File: internal/reader/ccip.go [hunk 4]
```diff
@@ -235,7 +234,10 @@ func (r *CCIPChainReader) CommitReportsGTETimestamp(
 		})
 	}
 
-	return reports, nil
+	if len(reports) < limit {
+		return reports, nil
+	}
+	return reports[:limit], nil
```
Comment (winder) [inline internal/reader/ccip.go:240]: If we aren't limiting in the query, should we remove the limit option and return everything?
With the current logic if we hit the limit with normal volume, I think we'd never get to some messages until things start timing out.

Comment (dimkouv) [reply internal/reader/ccip.go:247]: we'll fix those queries using proper contractReader args soon

**Review by 0xnogo — APPROVED** [review]

**Pull request merged**